docs/cluster_and_device_type_dev/cluster_and_device_type_dev.md

```diff
@@ -0,0 +1,248 @@
+# New Clusters & Device Types
+
+The goal of new cluster and device type development is to
+
+1. write the cluster implementations
+2. write the code and supporting material that will allow zap to generate the
+   appropriate ember layers
+3. write the unit tests, test plans and automation scripts that prove the code
+   correctness and allow these new features to be certified
+
+Unit tests, test plans and certification tests are covered in the testing
+section. This document will concentrate on implementing clusters and device
+types in the SDK.
+
+-   Cluster Definition
+    -   XML
+    -   Describes the structures, enums, attributes, commands, events etc.
+    -   Direct translation of the spec into code
+    -   src/app/zap-templates/zcl/data-model/chip/
+-   Cluster Implementation
+
+    -   Client side - codegen, you write the glue
+    -   Server side - cpp implementation through Ember and / or
+        AttributeAccessInterface & CommandHandlerInterface
+        -   src/app/clusters/<your_cluster_name>
+        -   build file: src/app/chip_data_model.gni
+        -   build file uses data from the codegen to auto-populate the cluster
+            list.
+        -   Follow examples in there to get your code building into the image
+            when selected in zap
+
+-   Device Type Definitions
+    -   XML defines conformance
+    -   src/app/zap-templates/zcl/data-model/chip/matter-devices.xml
+
+The following wiki page has a detailed description of how and where to add
+cluster and device type definitions so they are picked up properly by ZAP/ember
+and the SDK.
+
+[https://groups.csa-iot.org/wg/matter-tsg/wiki/howto-add-a-new-device-type](https://groups.csa-iot.org/wg/matter-tsg/wiki/howto-add-a-new-device-type)
```

Comment:
Should we consider translating this to adoc and have it here as well?

Reply:
At some point. That point is not today.

Reply:
Fair. Maybe I can find some time to look at that (at some point 😆)